files/image_config/monit/control_plane_drop_check

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+"""
+    This scripts's job is to verify that there are no control plane
+    packet drops reported by /proc/net/sofnet_stats.
+    This is to be run periodically on a SONiC device using a monit
+    configuration file.
+"""
+import os
+import sys
+import syslog
+
+
+def write_syslog(message, *args):
+    """
+    Write a message to syslog.
+
+    Args:
+        message (str): Message string to be logged
+        args: Optional args
+
+    Returns:
+        None
+    """
+
+    if args:
+        message %= args
+    syslog.syslog(syslog.LOG_NOTICE, message)
+
+
+def get_softnet_dropped_count():
+    """
+    Get dropped count from softnet stats procfs.
+
+    Returns:
+        drop_count (int): Number of dropped packets
+    """
+    drop_count = 0
+    softnet_stats_file = "/proc/net/softnet_stat"
+
+    with open(softnet_stats_file, 'r') as f:
+        for line in f:
+            if line.strip():  # Ensure the line is not empty
+                stat = line.split()
+                # Drop count is in the second column for each CPU
+                if len(stat) > 1:
+                    drop_count += int(stat[1], 16)
+
+    return drop_count
+
+
+def check_packet_drops():
+    """
+    The function that checks for kernel packet drops
+
+    Returns:
+        True if there are packet drops, False otherwise
+    """
+    # Path to the file to store the last drop count
+    drop_count_stash = '/tmp/softnet_dropped_count.txt'
+
+    # Read the last drop count if drop count stash file exists
+    if os.path.exists(drop_count_stash):
+        with open(drop_count_stash, 'r') as f:
+            count = f.read().strip()
+            last_drop_count = int(count) if count else 0
+    else:
+    # Initial run. Read drop count, update stash and return normally
+        last_drop_count = get_softnet_dropped_count()
+        with open(drop_count_stash, 'w') as f:
+            f.write(str(last_drop_count))
+        return False
```

Comment:
Any option to reset the last_drop_counter during config reload or warmboot??

Reply:
During warmboot, the counter will automatically be reset (after kexec to new kernel). For config reload case, we might have a stale value for potentially 1 cycle and then it will fix itself. The subsequent read will clear up the drop counter stash.

Reply:
Also verified that during "config reload", /proc/net/softnet_stat drop count does increase a bit, so we don't need to reset the last dropped counter after "config reload".

For eg: before config reload, the second column had a count of 0 and after config reload, the second column had a count of 00000015 (hex).